tests/Functional/SelfUpdateTest.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+/*
+ * This file is part of the Manalize project.
+ *
+ * (c) Manala <contact@manala.io>
+ *
+ * For the full copyright and license information, please refer to the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Manala\Manalize\Tests\Functional;
+
+use Manala\Manalize\Application;
+use Manala\Manalize\Command\Setup;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Process\Process;
+
+class SelfUpdateTest extends \PHPUnit_Framework_TestCase
+{
+    private static $cwd;
+
+    public function setUp()
+    {
+        $cwd = sys_get_temp_dir().'/Manala/tests/self-update';
+        $fs = new Filesystem();
+
+        if ($fs->exists($cwd)) {
+            $fs->remove($cwd);
+        }
+
+        $fs->mkdir($cwd);
+
+        self::$cwd = $cwd;
+    }
+
+    public function testExecute()
+    {
+        (new Process('make build', getcwd()))->run();
+        chmod(getcwd().'/manalize.phar', 0777);
```

Comment:
I think `MANALIZE_DIR.'/manalize.phar'` was the proper way, but maybe Travis is unable to build the phar because `phar.readonly` is true in it's configuration. You'll simply have to tweak it.

Also, is box2 installed ?